### PR TITLE
[xc-admin] improve approve/reject UX

### PR DESCRIPTION
**Pull request by cctdaniel**

### CODE REVIEW

**Review by guibescos — CHANGES_REQUESTED** [review]
There's a bug where if you click in a proposal and then back to proposals. The proposal disappears.

**Comment by cctdaniel** [comment]
> There's a bug where if you click in a proposal and then back to proposals. The proposal disappears.

can you try again?

File: governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals.tsx [hunk 1]
```diff
@@ -222,15 +222,18 @@ const Proposal = ({
...
 
   useEffect(() => {
-    // update the priceFeedMultisigProposals with previous value but replace the current proposal with the updated one
+    // update the priceFeedMultisigProposals with previous value but replace the current proposal with the updated one at the specific index
     if (currentProposal) {
       setpriceFeedMultisigProposals((prevProposals: TransactionAccount[]) => {
-        const proposals = prevProposals.filter(
-          (proposal) =>
-            proposal.publicKey.toBase58() !==
+        const index = prevProposals.findIndex(
```
Comment (guibescos) [inline governance/xc_admin/packages/xc_admin_frontend/components/tabs/Proposals.tsx:228]: Nitpick : can you use currentProposalIndex for this? It seems inefficient

**Review by guibescos — APPROVED** [review]
Seems to be working now!

**Pull request merged**